code/script.js

```diff
@@ -227,99 +258,166 @@ const setSecret = () => {
+  if (group === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person wears ${currentQuestion.attribute}! Keep all that wears ${currentQuestion.attribute}`
       )
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${currentQuestion.attribute}! Remove all that wears ${currentQuestion.attribute}`
       )
     }
-  } else if (category === 'other') {
-    // Similar to the one above
-  } else {
+  } else if (group === 'eye color') {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes the person has ${currentQuestion.value} eyes! Keep all with eyes ${currentQuestion.value}`
+      )
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert (
+        `No the person has not ${currentQuestion.value} eyes! Remove all with eyes ${currentQuestion.value}`
+      )
+    }
+  } else if (group === 'other') {
+    if (keep) { //attribute to show glasses, smoker and eye patch
+      alert(
+        `Yes, the person is a  ${currentQuestion.attribute}! Keep all persons who are a ${currentQuestion.attribute}`
+      )
+    } else {
+      alert (
+        `No, the person is not a ${currentQuestion.attribute}! Remove all the persons who are a ${currentQuestion.attribute}`
+      )
+    }
+    
+  } else {
+    if (keep) { 
+      alert (
+        `Yes, the person has ${currentQuestion.value} hair! Keep all with hair ${currentQuestion.value} `
+      )
+    } else {
+      alert (
+        `No, the person doesn't have ${currentQuestion.value} hair! Remove all with hair ${currentQuestion.value} `
+      )
+      
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+  if (keep) { //Ex: Does the person have glasses? If yes, they will be kept, if not, removed
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] === currentQuestion.value) 
+    //Keeps everyone who has got the asked attribute
+  } else {
+    charactersInPlay = charactersInPlay.filter((person) => person[currentQuestion.attribute] !== currentQuestion.value) 
+    //Removes everyone who doesn't have the asked attribute
+  }
```

Comment:
Maybe I'm not so good at reading code yet, haha so bare with me, but I'm getting confused with all the if (keep) statements. Could it be easier done as a function that you invoke after every alert? Or maybe not?